### ISSUE TITLE
[Code Idea]: Train with (or without) predictor.

### ISSUE THREAD

**Issue by galegozi** [lanl/Yoke#61]
## Summary

Instead of always using the predictor, train sometimes using just the encoder and decoder. The idea here is to try to ensure the following two statements are true (as frequently as possible):
1. When running the encoder then decoder on an image, the output is unchanged.
2. When running the full model it predicts outputs relatively well (including for chained prediction).

## Motivation

In current models, chained prediction is performing poorly.

## Affected Area

This would likely add a new training harness, but have little impact on the core code itself (if any).

## Proposed Approach

Train a new model that alternates between using and not using the predictor, adjusting the training script so that the loss is computed on self predictions when using the simplified model.

## Alternatives Considered

Although we could run chained predictions on the embeddings exclusively (#57), it would likely not be robust enough for real world use.

## Additional Context

Add any relevant information, discussions, related issues, or links.
